srcdeps-core/src/main/java/org/l2x6/srcdeps/core/config/Configuration.java

```diff
@@ -0,0 +1,169 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.l2x6.srcdeps.core.config;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.l2x6.srcdeps.core.BuildRequest.Verbosity;
+
+/**
+ * A configuration suitable for a build system wanting to handle source dependencies.
+ *
+ * @author <a href="https://github.com/ppalaga">Peter Palaga</a>
+ */
+public class Configuration {
+    public static String CONFIG_MODEL_VERSION = "1.0";
+
+    public static class Builder {
+
+        private Set<String> forwardProperties = new LinkedHashSet<>();
+        private BuilderIo builderIo;
+        private List<ScmRepository> repositories = new ArrayList<>();
+        private boolean skip = false;
+        private boolean skipTests = true;
+        private Path sourcesDirectory;
+        private Verbosity verbosity = Verbosity.warn;
+
+        private Builder() {
+            super();
+        }
+
+        public Configuration build() {
+            return new Configuration(repositories, sourcesDirectory, skipTests, skip, verbosity, builderIo,
+                    forwardProperties);
+        }
+
+        public Builder forwardProperty(String value) {
+            forwardProperties.add(value);
+            return this;
+        }
+
+        public Builder forwardProperties(Collection<String> values) {
+            forwardProperties.addAll(values);
+            return this;
+        }
+
+        public Builder builderIo(BuilderIo.Builder builderIo) {
+            this.builderIo = builderIo.build();
+            return this;
+        }
+
+        public Builder skip(boolean value) {
+            this.skip = value;
+            return this;
+        }
+
+        public Builder skipTests(boolean value) {
+            this.skipTests = value;
+            return this;
+        }
+
+        public Builder sourcesDirectory(Path value) {
+            this.sourcesDirectory = value;
+            return this;
+        }
+
+        public Builder verbosity(Verbosity value) {
+            this.verbosity = value;
+            return this;
+        }
+
+        public Builder repositories(Map<String, ScmRepository.Builder> repoBuilders) {
+            for (Map.Entry<String, ScmRepository.Builder> en : repoBuilders.entrySet()) {
+                ScmRepository.Builder repoBuilder = en.getValue();
+                repoBuilder.id(en.getKey());
+                this.repositories.add(repoBuilder.build());
+            }
+            return this;
+        }
+
+        public Builder configModelVersion(String configModelVersion) {
+            if (!CONFIG_MODEL_VERSION.equals(configModelVersion)) {
```

Comment:
`configModelVersion` exists just in case we'll have to change the schema of srcdeps.yaml in the future. `1.0` is the only possible version ATM. <- this text is in srcdeps.yaml.adoc now https://github.com/ppalaga/srcdeps/blob/d09cac5775e39e99c23f7b97bf6497d52a057fc1/srcdeps.yaml.adoc#mvnsrcdepsyaml 

I think that any forward compatibility depends on the nature of the future changes. As far as possible I'd try to have two (or multiple) parallel schemata in the future rather than replacing the old with new ones. The `configModelVersion` marker is there exactly for that: to be able to choose the right parser for the given file. Does that sound well?